src/common/blkdev.cc

```diff
@@ -222,6 +222,52 @@ bool BlkDev::is_rotational() const
   return get_int_property("queue/rotational") > 0;
 }
 
+int get_device_num_luns(const std::string& path, std::string *err)
```

Comment:
Might we avoid the dependency of sg_lun and the associated screen scraping of it's output by leveraging:
/dev/disk/by-path/  
where softlink names are decorated with the LUN are available and the softlink itself points back to the kernel scsi device in /dev.

An example of how I did this in bash is in the Appendix as the script "mk_lvm.sh" in this document:
https://docs.google.com/document/d/14HzzezBJKB1DoHD8G3sYYzx-rIMRUnrnizwSeGBa-gU/edit?usp=sharing

Reply:
YES--that would be so much better.  will revise.